python/lsst/pipe/tasks/matchBackgrounds.py

```diff
@@ -19,683 +19,627 @@
-                                               | afwMath.NPOINT | afwMath.STDEV,
-                                               self.sctrl)
-                npoints, _ = stats.getResult(afwMath.NPOINT)
-                if npoints >= 2:
-                    stdev, _ = stats.getResult(afwMath.STDEV)
-                    if stdev < self.config.gridStdevEpsilon:
-                        stdev = self.config.gridStdevEpsilon
-                    bgX.append(0.5 * (x0 + xmin + x0 + xmax))
-                    bgY.append(0.5 * (y0 + ymin + y0 + ymax))
-                    bgdZ.append(stdev/numpy.sqrt(npoints))
-                    est, _ = stats.getResult(statsFlag)
-                    bgZ.append(est)
-
-        return numpy.array(bgX), numpy.array(bgY), numpy.array(bgZ), numpy.array(bgdZ)
+        # TODO: leaving as-is for now, but currently this only fits the pre-
+        # binned images, meaning binSize is always 1.  But we need a solution
+        # to the 0 variance image issue, which might mean we need a variable
+        # binSize still.
+        nx = warp.getWidth() // binSize
+        ny = warp.getHeight() // binSize
 
+        bgCtrl = BackgroundControl(nx, ny, self.statsCtrl, self.statsFlag)
+        bgCtrl.setUndersampleStyle(self.config.undersampleStyle)
+        bkgd = makeBackground(warp, bgCtrl)
 
-class DataRefMatcher:
-    """Match data references for a specified dataset type.
+        return bkgd, bgCtrl
 
-    Note that this is not exact, but should suffice for this task
-    until there is better support for this kind of thing in the butler.
-
-    Parameters
-    ----------
-    butler : `lsst.daf.butler.Butler`
-        Butler to search for maches in.
-    datasetType : `str`
-        Dataset type to match.
-    """
-
-    def __init__(self, butler, datasetType):
-        self._datasetType = datasetType  # for diagnostics
-        self._keyNames = butler.getKeys(datasetType)
-
-    def _makeKey(self, ref):
-        """Return a tuple of values for the specified keyNames.
+    def _fluxScale(self, exposure):
+        """Scales image to nJy flux using photometric calibration.
 
         Parameters
         ----------
-        ref : `Unknown`
-            Data reference.
+        exposure: `lsst.afw.image._exposure.ExposureF`
+            Exposure to scale.
 
-        Raises
-        ------
-        KeyError
-            Raised if ref.dataId is missing a key in keyNames.
+        Returns
+        -------
+        fluxZp: `float`
+            Counts to nanojanskies conversion factor
         """
-        return tuple(ref.dataId[key] for key in self._keyNames)
-
-    def isMatch(self, ref0, ref1):
-        """Return True if ref0 == ref1.
-
-        Parameters
-        ----------
-        ref0 : `Unknown`
-            Data for ref 0.
-        ref1 : `Unknown`
-            Data for ref 1.
+        fluxZp = exposure.getPhotoCalib().instFluxToNanojansky(1)
+        exposure.image *= fluxZp
+        return fluxZp
```

Comment:
Since warps are now nJy, I don't think you need this method anymore, and deleting it is the easiest way to find the places that call it.